### PR TITLE
Add Assign Impedance Boundary to Maxwell 3D.

### LINKED ISSUE #1488: Add Assign Impedance Boundary to Maxwell 3D
### Description of the feature

Impedance Boundary is pretty commonly used to model conductive objects at higher frequencies, where the skin depth is small. This is applicable for Frequency Domain (Eddy Current) designs in Maxwell 3D.

### Steps for implementing the feature

Implementation of existing Maxwell 3D API:

oDesign.SetSolutionType("EddyCurrent")
oModule = oDesign.GetModule("BoundarySetup")
oModule.AssignImpedance(
	[
		"NAME:Impedance1",
		"Faces:="		, [35],
		"UseMaterial:="		, False,
		"Permeability:="	, "1",
		"Conductivity:="	, "1"
	])

List of faces, permeability and conductivity should be inputs.

### Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by MaxJPRey**
Fix #1488 .

### CODE REVIEW

File: pyaedt/maxwell.py [hunk 1]
```diff
@@ -1460,15 +1460,104 @@ def assign_insulating(self, geometry_selection, insulation=None):
...
 
-            return self._create_boundary(insulation, props, "Insulating")
+            return self._create_boundary(insulation_name, props, "Insulating")
+        return False
+
+    @pyaedt_function_handler()
+    def assign_impedance(
+        self,
+        geometry_selection,
+        use_material=False,
```
Comment (maxcapodi78) [inline pyaedt/maxwell.py:1478]: @MaxJPRey what if we remove this boolean and simply check if material_name is passed and automatically set internally to True or False depending if the material is passed or not?

**Review by maxcapodi78 — APPROVED** [review]
Apart a minor comment it LGTM

File: pyaedt/maxwell.py (hunk 1 above)
Comment (MaxJPRey) [reply pyaedt/maxwell.py:1478]: That is an excellent idea. I will implement that.

**Review by maxcapodi78 — APPROVED** [review]
LGTM

**Pull request merged**